### PR TITLE
'mix archive.build' bundles dot files from priv folder

### PR DESCRIPTION
**Pull request by predrag-rakic**

### CODE REVIEW

**Comment by predrag-rakic** [comment]
`make clean test` passes locally.

Build failed with message: 
> No output has been received in the last 10m0s, this potentially indicates a stalled build ...

Will close and reopen PR to initiate rebuild.

**Comment by predrag-rakic** [comment]
Reopening to initiate rebuild.

**Comment by josevalim** [comment]
Thank you but I am a bit torn on this one because you may also have editor and other lingering files that will pollute the archive. What is the use case for this? Maybe we should hide this behind a flag?

**Comment by predrag-rakic** [comment]
We have a tool that calls `mix new` and adds more files to created folder (among other things).
Some of the added files are dot files like `.gitignore`.

I'm fine with hiding the feature behind the flag.

**Comment by josevalim** [comment]
@predrag-rakic so let's go with the flag approach. We can call it `--include-dot-files` or similar. Thanks!

**Comment by predrag-rakic** [comment]
@josevalim `--include-dot-files` flag implemented.

File: lib/mix/lib/mix/tasks/archive.build.ex [hunk 1]
```diff
@@ -84,26 +87,26 @@ defmodule Mix.Tasks.Archive.Build do
...
 
-    create(source, target)
+    create(source, target, Keyword.get(opts, :include_dot_files, false))
 
     Mix.shell.info "Generated archive #{inspect target} with MIX_ENV=#{Mix.env}"
     :ok
   end
 
-  defp create(source, target) do
+  defp create(source, target, include_dot_files) do
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/archive.build.ex]: Should we call this variable `include_dot_files?` since it is a boolean? :)

Comment (josevalim) [inline lib/mix/lib/mix/tasks/archive.build.ex]: This and the occurrences below.

**Comment by predrag-rakic** [comment]
@josevalim Tests failed again for some weird reason. Do you want me to rebuild until it passes?

**Comment by josevalim** [comment]
@predrag-rakic i will do that on travis, thanks!

**Comment by predrag-rakic** [comment]
@josevalim  Thank you.

As a side note, I work for SemaphoreCI and as such I might not be the most objective person in the world, but having 2 false negative builds out of 5 is bad UX.

**Comment by josevalim** [comment]
@predrag-rakic definitely. This last one was our fault though - a test for a recently added feature that would fail depending on the seed of the test suite.

File: lib/mix/test/mix/tasks/archive_test.exs [hunk 2]
```diff
@@ -25,14 +25,26 @@ defmodule Mix.Tasks.ArchiveTest do
   test "archive build" do
     in_fixture "archive", fn ->
       Mix.Tasks.Archive.Build.run ["--no-elixir-version-check"]
-      assert File.regular? 'archive-0.1.0.ez'
-      assert has_zip_file?('archive-0.1.0.ez', 'archive-0.1.0/.elixir')
-      assert has_zip_file?('archive-0.1.0.ez', 'archive-0.1.0/priv/not_really_an.so')
-      assert has_zip_file?('archive-0.1.0.ez', 'archive-0.1.0/ebin/Elixir.Mix.Tasks.Local.Sample.beam')
-      assert has_zip_file?('archive-0.1.0.ez', 'archive-0.1.0/ebin/archive.app')
+      assert_archive_content_default()
```
Comment (ericmj) [inline lib/mix/test/mix/tasks/archive_test.exs]: We should add a `refute has_zip_file?('archive-0.1.0.ez', 'archive-0.1.0/priv/.dot_file')` here.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**